artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ReplicationPrimaryActivation.java

```diff
@@ -0,0 +1,489 @@
+    */
+   private void awaitBackupAnnouncementOnFailbackRequest(ClusterConnection clusterConnection) throws Exception {
+      final String nodeID = activeMQServer.getNodeID().toString();
+      final BackupTopologyListener topologyListener = new BackupTopologyListener(nodeID, clusterConnection.getConnector());
+      clusterConnection.addClusterTopologyListener(topologyListener);
+      try {
+         if (topologyListener.waitForBackup()) {
+            restartAsBackupAfterFailback();
+         } else {
+            ActiveMQServerLogger.LOGGER.failbackMissedBackupAnnouncement();
+         }
+      } finally {
+         clusterConnection.removeClusterTopologyListener(topologyListener);
+      }
+   }
+
+   /**
+    * If {@link #asyncStopServer()} happens before this call, the restart just won't happen.
+    * If {@link #asyncStopServer()} happens after this call, will make the server to stop right after being restarted.
+    */
+   private void restartAsBackupAfterFailback() throws Exception {
+      if (stoppingServer) {
+         return;
+      }
+      synchronized (this) {
+         if (stoppingServer) {
+            return;
+         }
+         final String coordinatedLockAndNodeId = activeMQServer.getNodeManager().getNodeId().toString();
+         final long inSyncReplicaActivation = activeMQServer.getNodeManager().getNodeActivationSequence();
+         DistributedLock existingLiveLock = distributedManager.getDistributedLock(coordinatedLockAndNodeId);
+         // give up our role as 'live' asap, will also happen on server.fail
+         existingLiveLock.close();
+
+         activeMQServer.fail(true);
+
+         // need to restart
+         distributedManager.start();
+         final MutableLong coordinatedActivationSequence = distributedManager.getMutableLong(coordinatedLockAndNodeId);
+         // wait for the live to activate and run un replicated with a sequence > inSyncReplicaActivation
+         // this read can be dirty b/c we are just looking for an increment, we don't care about the actual value
+         final long done = System.currentTimeMillis() + policy.getBackupPolicy().getVoteRetryWait();
+         do {
+            final long coordinatedValue = coordinatedActivationSequence.get();
+            if (coordinatedValue > inSyncReplicaActivation) {
+               // all good, activation has gone ahead
+               LOGGER.infof("Detected expected sequential server activation after failback, with NodeID = %s: and sequence: %d", coordinatedLockAndNodeId, coordinatedValue);
+               break;
+            }
+            try {
+               TimeUnit.MILLISECONDS.sleep(100);
+            } catch (InterruptedException ignored) {
+            }
+         }
+         while (done < System.currentTimeMillis());
+
+         if (coordinatedActivationSequence.get() == inSyncReplicaActivation) {
+            LOGGER.warnf("Timed out waiting for failback server activation with NodeID = %s: and sequence > %d: after %dms",
```

Comment:
what happen if the other primary hasn't been able to failback?

If this broker is going to restart as backup it will skip checking any activation sequence and will rotate the data (while preserving the NodeID), is it correct?
In the worst case scenario the other primary has died and there is no live around AND this broker will rotate the most up to date data, that seems a bad thing. Let's speak on monday about it: maybe makes sense to NOT skip checking the activation sequence on server restart during a failback, in short...
```java
         if (policy.isTryFailback()) {
            // we are replicating to overwrite our data, transient backup state while trying to be the primary
         } else {
            // we may be a valid insync_replica (backup) if our activation sequence is largest for a nodeId
            // verify that before removing data..
            final DistributedLock liveLockWithInSyncReplica = checkForInSyncReplica();
```
can become
```java
            // we may be a valid insync_replica (backup) if our activation sequence is largest for a nodeId
            // verify that before removing data..
            final DistributedLock liveLockWithInSyncReplica = checkForInSyncReplica();
```
because `checkForInSyncReplica` won't give up on becoming backup if the data is the most up to date.
wdyt?

Reply:
Just a side note: on primary's start it search in the topology for a live server and if found, it restart as a (ready to failback) backup: if we `checkForInSyncReplica` it should be fine, because if there's another live server with its acceptor already opened, it means that it already own the live lock and has successfully increased activation sequence, hence `checkForInSyncReplica` should be able to fail fast thanks to the dirty read introduced recently and just make the failback process to start as expected.